### ISSUE TITLE
getting DPY-4000 in thin mode, add support for comma separated `net_service_name`s in tnsnames.ora

### ISSUE THREAD

**Issue by izirku** [oracle/python-oracledb#362]
In our environment, `tnsname.ora` has many entries of form:

```
service_alias1.domain,
service_alias2.domain =
  (DESCRIPTION
    ...
  )
```

This works fine with `sqlplus`, many clients, and `python-oracledb` in **thick mode**. However, it doesn't work in **thin mode** which is required by `python-oracledb` for concurrent APIs.

The problem part is `net_service_alias1, net_service_alias2 =`, it causes a `DPY-4000: unable to find "net_service_alias1" in /opt/instantclient/network/admin/tnsnames.ora` error when trying to resolve `net_service_alias1`. It seems that `python-oracledb` doesn't support a list of comma separated `net_service_name` values.

A temporary workaround, creating "shadow" copy of `tnsname.ora` and removing second `net_service_name`:

```
service_alias1.domain =
  (DESCRIPTION
    ...
  )
...
```

, makes thin mode name resolution work. Please let me know if more details are needed! This enhancement would be a great improvement, thanks in advance!

Platform:

- RHEL9, other Linux x86_64 systems
- Python 3.12.4
- `python-oracledb` 2.2.1

**Comment by cjbj** [comment]
This was fixed in https://github.com/oracle/python-oracledb/issues/3#issuecomment-1139806821 and should work.

In fact, I can't reproduce your problem.  Is there some interplay of domains, or special characters that causes it to reproduce for you?

I tried `/tmp/tnsnames.ora`:
```
service_alias1.domain,
service_alias2.domain,
mydb.x = (DESCRIPTION=(ADDRESS=(PROTOCOL=TCP)(HOST=localhost)(PORT=1521))(CONNECT_DATA=(SERVER=DEDICATED)(SERVICE_NAME=orclpdb1)))
```
and the code:
``` 
    connection = oracledb.connect(user=un, password=pw, config_dir='/tmp', dsn='mydb.x')

    with connection.cursor() as cursor:
        sql = """select systimestamp from dual"""
        for (r,) in cursor.execute(sql):
            print(r)
```

This connected and ran successfully.

**Comment by izirku** [comment]
Ok, I think, I see what's going on, only the alias closest to the `=` / on the same as `=` gets resolved. Could you please try running it with `/tmp/tnsnames.ora`, but move `mydb.x` to a first line, and try to resolve for `mydb.x`?

My invocation of `oracledb.connect()` is exact same as yours otherwise. I just ran a couple of similar tests, and these are the findings on my end (in thin client mode):

this form doesn't resolve for `mydb.x`:

```
mydb.x,
service_alias1.domain,
service_alias2.domain = (DESCRIPTION=(ADDRESS=(PROTOCOL=TCP)(HOST=localhost)(PORT=1521))(CONNECT_DATA=(SERVER=DEDICATED) SERVICE_NAME=orclpdb1)))
```

while this one does resolve for `mydb.x` (which is not how autogenerated `tnsnames.ora` looks, i.e., all aliases are on separate lines when autogenerated):

```
mydb.x, service_alias1.domain, service_alias2.domain =
  (DESCRIPTION=(ADDRESS=(PROTOCOL=TCP)(HOST=localhost)(PORT=1521))(CONNECT_DATA=(SERVER=DEDICATED)(SERVICE_NAME=orclpdb1)))
```

Also, thank you for looking into this!

**Comment by cjbj** [comment]
Yep confirmed.  Thanks for doing the extra investigation.

**Comment by anthony-tuininga** [comment]
I have pushed a patch that corrects this issue. If you are able to [build from source](https://python-oracledb.readthedocs.io/en/latest/user_guide/installation.html#installing-from-source-code) you can verify that it works for you, too.

**Comment by anthony-tuininga** [comment]
This was included in python-oracledb 2.4.0 which was just released.
